src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCache.cs

```diff
@@ -79,12 +79,7 @@ public SimpleQueueCache(int cacheSize, Logger logger)
         /// </summary>
         public virtual bool IsUnderPressure()
         {
-            if (cachedMessages.Count == 0) return false; // empty cache
-            if (Size < maxCacheSize) return false; // there is still space in cache
-            if (cacheCursorHistogram.Count == 0) return false;    // no cursors yet - zero consumers basically yet.
-            // cache is full. Check how many cursors we have in the oldest bucket.
-            int numCursorsInLastBucket = cacheCursorHistogram[0].NumCurrentCursors;
-            return numCursorsInLastBucket > 0;
+            return cacheCursorHistogram.Count >= NUM_CACHE_HISTOGRAM_BUCKETS;
```

Comment:
Can u please explain why u had to change that function and how it will work now? Why the current code is correct? Not saying it's not, just want u to explain the rationale and how it will work now.